packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -128,6 +132,25 @@ export function SummaryContainer() {
     );
   });
 
+  /** useEffect to clear projects from expired rounds (no longer accepting donations) */
+  useEffect(() => {
+    if (!rounds) {
+      return;
+    }
+    /*get rounds that have expired */
+    const expiredRounds = rounds
+      .filter((round) => round.roundEndTime.getTime() < Date.now())
+      .map((round) => round.id)
+      .filter((id): id is string => id !== undefined);
```

Comment:
Hmm, this is a mixed message, because if `id is string` then `id` cannot be undefined. I guess that you're trying to get the compiler to infer proper types.

Would this work?

```ts
import { isPresent } from 'ts-is-present';
// ...
  .map((round) => round.id)
  .filter(isPresent)
```

Also, I'd name the variable `expiredRoundIds` to clarify that it's just the ids.

Reply:
Sorry, I realized I misread the type assertion as cast, disregard the "mixed messages" part. (`isPresent` might still be worth a try though.)

Reply:
I forgot we have isPresent already as a dep, fixed.